modules/exploits/linux/local/omniresolve_suid_priv_esc.rb

```diff
@@ -0,0 +1,124 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Local
+  Rank = ExcellentRanking
+
+  include Msf::Post::File
+  include Msf::Post::Linux::Priv
+  include Msf::Post::Linux::System
+  include Msf::Exploit::EXE
+  include Msf::Exploit::FileDropper
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'           => 'Micro Focus (HPE) Data Protector SUID Privilege Escalation',
+      'Description'    => %q{
+      This module exploits the trusted $PATH environment variable of the SUID binary omniresolve.
+
+      This module has been successfully tested on:
+      HPE Data Protector A.09.07: OMNIRESOLVE, internal build 110, built on Thu Aug 11 14:52:38 2016
+      Micro Focus Data Protector A.10.40: OMNIRESOLVE, internal build 118, built on Tue May 21 05:49:04 2019 on CentOS Linux release 7.6.1810 (Core)
+
+      The vulnerability has been patched in:
+      Micro Focus Data Protector A.10.40: OMNIRESOLVE, internal build 125, built on Mon Aug 19 19:22:20 2019
+      },
+      'License'        => MSF_LICENSE,
+      'Author'         =>
+        [
+          's7u55', # Discovery and Metasploit module
+        ],
+      'DisclosureDate' => '2019-09-13',
+      'Platform'       => [ 'linux' ],
+      'Arch'           => [ ARCH_X86, ARCH_X64 ],
```

Comment:
Does Micro Focus (HPE) Data Protector run on multiple platforms? If so, the current best practice, as stupid as it looks, is to do:

```ruby
      'Arch'           =>
        [
          ARCH_X86,
          ARCH_X64,
          ARCH_ARMLE,
          ARCH_AARCH64,
          ARCH_PPC,
          ARCH_MIPSLE,
          ARCH_MIPSBE
        ],
```

Reply:
It supports HP-UX platform, but I have not tested that.